test/e2e_federation/framework/framework.go

```diff
@@ -40,33 +40,40 @@ import (
 type Framework struct {
 	*framework.Framework
 
+	// To make sure that this framework cleans up after itself, no matter what,
+	// we install a Cleanup action before each test and clear it after.  If we
+	// should abort, the AfterSuite hook should run all Cleanup actions.
+	cleanupHandle framework.CleanupActionHandle
+
 	FederationClientset *federation_clientset.Clientset
 	FederationNamespace *v1.Namespace
 }
 
 func NewDefaultFederatedFramework(baseName string) *Framework {
-	options := framework.FrameworkOptions{
-		ClientQPS:   20,
-		ClientBurst: 50,
-	}
+	f := &Framework{}
+
+	// Register the federation cleanup before initializing the default
+	// e2e framework to ensure it gets called before the default
+	// framework's cleanup.
+	AfterEach(f.FederationAfterEach)
 
-	f := &Framework{&framework.Framework{
-		BaseName:                 baseName,
-		AddonResourceConstraints: make(map[string]framework.ResourceConstraint),
-		Options:                  options,
-		ClientSet:                nil,
-	}, nil, &v1.Namespace{}}
+	f.Framework = framework.NewDefaultFramework(baseName)
```

Comment:
ugh why does &Framework{} have a Framework ? weird . but i think it makes sense.